lms/djangoapps/grades/tasks.py

```diff
@@ -19,25 +21,51 @@
 from .signals.signals import SUBSECTION_SCORE_CHANGED
 from .transformer import GradesTransformer
 
+log = getLogger(__name__)
+
 
 @task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
-def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher):
+def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher, grade, max_grade):
     """
     Updates a saved subsection grade.
     This method expects the following parameters:
-       - user_id: serialized id of applicable User object
-       - course_id: Unicode string representing the course
-       - usage_id: Unicode string indicating the courseware instance
-       - only_if_higher: boolean indicating whether grades should
+        - user_id: serialized id of applicable User object
+        - course_id: Unicode string representing the course
+        - usage_id: Unicode string indicating the courseware instance
+        - only_if_higher: boolean indicating whether grades should
         be updated only if the new grade is higher than the previous
         value.
+        - grade: the points the learner earned on the problem that
+        triggered the update
+        - max_grade: the max points the leaner could have earned
+        on the problem
     """
     if not PersistentGradesEnabledFlag.feature_enabled(course_id):
         return
 
     course_key = CourseLocator.from_string(course_id)
     student = User.objects.get(id=user_id)
     scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    student_module = StudentModule.objects.get(
```

Comment:
Can we call `get_score` in model_data.py instead of accessing the data model object directly?